### ISSUE TITLE
Floating label doesn't work on input[type=email] when value is not valid

### ISSUE THREAD

**Issue by eliashdezr** [quasarframework/quasar#237]
Using this code 

```
<div class="floating-label">
  <input required="required" type="email" class="full-width">
<label>Email</label></div>
```

And typing  anything without the `@` causes this behavior on the label:
![2016-12-06 23_12_46-asistente asesores](https://cloud.githubusercontent.com/assets/6153607/20955531/8b228510-bc09-11e6-8f51-cac86c3b078d.png)

**Comment by vuchl** [comment]
Hmm that's kind of strange. from https://github.com/quasarframework/quasar/blob/dev/src/css-components/textfield/textfield.mat.styl and https://github.com/quasarframework/quasar/blob/dev/src/css-components/textfield/textfield.ios.styl there is not difference in the types of `input` tags to be seen. But maybe I'm looking at the wrong part of the code.

**Comment by rstoenescu** [comment]
New component coming soon for floating label.

**Comment by eliashdezr** [comment]
@rstoenescu Cool!

Btw, just wanted to add another strange behavior on the password input label:
![2016-12-07 04_07_13-asistente asesores](https://cloud.githubusercontent.com/assets/6153607/20963391/c9125d22-bc32-11e6-9a7c-007bb62a1171.png)

Hope this helps in the new component!

**Comment by CookieJon** [comment]
@eliashdezr  Thanks, I'll make sure that's included in the fix (which I'm doing now)

**Comment by rstoenescu** [comment]
PR is being worked on by @CookieJon

**Comment by oswaldomilet** [comment]
@CookieJon @rstoenescu Don't know if you found the solution, I guess the problem happens here:

https://github.com/quasarframework/quasar/blob/dev/src/css-components/textfield/textfield.mat.styl#L67 (`input:valid`)

I was investigating how MaterializeCss solve this, they put a .active class via jQuery into input when it doesn't have focus or when it have a value.

@eliashdezr as a workahound for now you could remove `type="email"` from input and validate it with [vuelidate](http://quasar-framework.org/guide/form-validation.html) (it has an email validator out of the box)

**Comment by rstoenescu** [comment]
Consider this solved. There's a mega-patch coming up for form components.

**Comment by oswaldomilet** [comment]
Do you have a planned date for this to be released?

**Comment by rstoenescu** [comment]
@oswaldomilet It's exactly what I've been working on the whole week :) + more other form related things.

**Comment by rstoenescu** [comment]
Will be available in v0.14
